gitpod-io/gitpod
# [dashboard] Onboarding: Add Gitpod repository template to New workspace component

## Description


This PR adds a Gitpod repository template to the New workspace component. The template allows users to easily create workspaces with preconfigured demos for Docker Compose and Java with Maven and Spring Boot.

It also adds a new `showExamples` option to `/new` (as `/new?showExmaples=true`) `CreateWorkspacePage` component. The option allows controlling whether example repositories should be shown.


https://github.com/gitpod-io/gitpod/assets/55068936/543bb1e5-6808-4d1e-ae11-9d687738ac13



### RE: Tracking

It already exist in past, so didn't changed anything, for these repos, we can use either of the method track:

- No. of workspaces started from those two github repositories (can be slightly incorrect, because repos. are publicly accessible via GitHub, Website docs, etc.)
- Measuring `dashboard_clicked` event following labels:
    - label: `demo-javagithub.com/gitpod-demos/spring-petclinic` _(label names are broken since before)_
    - label: `voting-appgithub.com/gitpod-demos/voting-app`


## Related Issue(s)

Fixes OPM-656

## How to test


- Open preview environment
- Complete onboarding
- Click on `New Workspace` (`/new`)
- Scroll down to the `repository` list and see :eyes:
- Same for `/new?showExamples=true`

## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - siddhant-offf79390ef</li>
	<li><b>🔗 URL</b> - <a href="https://siddhant-offf79390ef.preview.gitpod-dev.com/workspaces" target="_blank">siddhant-offf79390ef.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - siddhant-opm-656-add-demo-projects-to-repository-select-on-new-workspace-page-gha.27021</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-siddhant-offf79390ef%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
